packages/eslint-plugin/docs/rules/await-thenable.mdx

```diff
@@ -40,6 +40,76 @@ await createValue();
 </TabItem>
 </Tabs>
 
+## Async Iteration (`for await...of` Loops)
+
+This rule also inspects [`for await...of` statements](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for-await...of), which are designed for iterating over async-iterable objects.
+If the value being iterated over is not async-iterable, an ordinary `for...of` statement is preferable, even if the value is an iterable of Thenables.
```

Comment:
This is debatable. Banning iterables of thenables should be an option.

Reply:
Eh, kind of... in any case, I was going off of https://github.com/typescript-eslint/typescript-eslint/issues/8858#issuecomment-2148977352. It's easy enough to add an option to allow iterables of thenables, though I'm -0.5 on doing so.

Reply:
In lieu of an option, can we add more motivation to the docs (as what I said in that comment)?

Reply:
Good point! Made some changes around this 👍